alphagov/paas-cf
# [#110953182] Replace s3get with aws-cli

## What

This is some "gardening" preparation for the referenced story and to prevent other re-use of this script. I'm not doing the other work described in the story.

#### README: Replace use of s3get with aws-cli

We can use the official aws-cli tool to do the same job as `s3get.sh`. If we
also replace the use of the script in the pipeline, then we can remove this
script from our repo.

I've included a very brief description (with some links) about how to
install and configure it.

#### create-deployer: Replace s3get with aws-cli

Use the official aws-cli utility instead of our custom script. We hope to
replace this task with s3-resource, if we can add support for it to error
instead of block when the bucket doesn't exist:

- https://www.pivotaltracker.com/story/show/110953182

In the meantime though, replacing this means that we have less code to
maintain. This is especially important as we don't currently have any tests
for this functionality.

We originally created the aws-cli container for 5448cee.

#### try-fetch-bucket-state: explicit error handling

Add the `-e` flag to the `try-fetch-bucket-state` task to exit early on
errors, as used in all other tasks. Then add `|| true` to the command which
attempts to fetch the state from the bucket and we expect to fail on the
very first run. This has the same effect but makes error handling more
explicit to read.

## How to test

1. Destroy your existing Deployer Concourse (and BOSH/CF) or choose a new environment name, so that you can test this from scratch
1. Run the `create-deployer` pipeline and check that it completes successfully
1. Run the `create-deployer` pipeline again and check that it completes successfully

## Who can review

Not @dcarley
